netbout-web/src/main/java/com/netbout/rest/TkStart.java

```diff
@@ -67,13 +75,45 @@ public TkStart(final Base bse) {
 
     @Override
     public Response act(final Request req) throws IOException {
-        final Inbox inbox = new RqAlias(this.base, req).alias().inbox();
-        final long number = inbox.start();
+        final Href href = new RqHref.Base(req).href();
+        final Iterator<String> token = href.param("token").iterator();
+        String tokenkey = "";
+        final Inbox inbox;
+        long number = 0;
+        while (token.hasNext()) {
+            tokenkey = token.next();
+        }
+        if (tokenkey.equals("")) {
+            inbox = new RqAlias(this.base, req).alias().inbox();
+        } else {
+            if (this.tokencache.containsKey(tokenkey)) {
```

Comment:
@prahladyeri I believe this `if-else` statement doesn't have to be nested, and can be moved to the same level as the outer `else`:

```java
if (tokenkey.equals("")) {
    tokenkey = token.next();
} else if (this.tokencache.containskey(tokenkey)) {
 // truncated code in inner if block
} else {
  // truncated code in inner else block
}
```